homeassistant/components/trafikverket_weatherstation/sensor.py

```diff
@@ -191,26 +191,26 @@ def __init__(
             name=sensor_station,
             configuration_url="https://api.trafikinfo.trafikverket.se/",
         )
-        self._weather: WeatherStationInfo | None = None
-
-    @Throttle(MIN_TIME_BETWEEN_UPDATES)
-    async def async_update(self) -> None:
-        """Get the latest data from Trafikverket and updates the states."""
-        try:
-            self._weather = await self._weather_api.async_get_weather(self._station)
-        except (asyncio.TimeoutError, aiohttp.ClientError, ValueError) as error:
-            _LOGGER.error("Could not fetch weather data: %s", error)
-            return
-        self._attr_native_value = getattr(
-            self._weather, self.entity_description.api_key
+
+    @property
+    def native_value(self) -> StateType:
+        """Return state of sensor."""
+        state: StateType = getattr(
+            self.coordinator.data, self.entity_description.api_key
         )
-        if (
-            self._attr_native_value is None
-            and self.entity_description.key in NONE_IS_ZERO_SENSORS
-        ):
-            self._attr_native_value = 0
-
-        self._attr_extra_state_attributes = {
-            ATTR_ACTIVE: self._weather.active,
-            ATTR_MEASURE_TIME: self._weather.measure_time,
+        # For zero value state the api reports back None for certain sensors.
+        if state is None and self.entity_description.key in NONE_IS_ZERO_SENSORS:
+            return 0
+        return state
+
+    @property
+    def extra_state_attributes(self) -> dict[str, str]:
+        """Return extra attributes."""
+        return {
+            ATTR_MEASURE_TIME: _to_iso_format(self.coordinator.data.measure_time),
         }
```

Comment:
If these update frequently we should make this its own sensor

Reply:
It's not that frequent but still I agree so I move this to own sensor